addons/sale_stock/models/sale_order.py

```diff
@@ -356,6 +358,35 @@ def _compute_is_mto(self):
             else:
                 line.is_mto = False
 
+    @api.depends('move_ids')
+    def _compute_json_forecast(self):
+        self.json_forecast = False
+        if not any(self._ids):
+            # onchange
+            return
+        # compute
+        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        for so_line in self:
+            if any(so_line.move_ids.filtered(lambda move: move.state in ['done'])):
+                qty_delivered = float_repr(so_line.qty_delivered, precision)
+                so_line.json_forecast = json.dumps({'reservedAvailability': qty_delivered})
+            else:
+                # For waiting deliveries, take the info from the report line.
+                delivery = so_line.move_ids.picking_id.filtered(lambda picking: picking.state in ['confirmed', 'waiting'])
+                if delivery.exists():
+                    moves_to_process = so_line.move_ids.filtered(lambda move: move.id in delivery.move_line_ids.ids and move.state not in ['cancel', 'done'])
+                    if moves_to_process.exists():
+                        so_line.json_forecast = moves_to_process[0].json_forecast
+                else:
+                    # For assigned deliveries, take the delivery's date.
+                    delivery = so_line.move_ids.picking_id.filtered(lambda picking: picking.state == 'assigned')
+                    if delivery.exists():
+                        date_expected = delivery.scheduled_date
+                        so_line.json_forecast = json.dumps({
+                            'expectedDate': format_date(self.env, date_expected),
+                            'isLate': date_expected > so_line.order_id.expected_date,
```

Comment:
order_id.expected_date missing in the depends

Reply:
I need to add `qty_delivered` too, isn't ?

Reply:
yep